pinot-spi/src/main/java/org/apache/pinot/spi/accounting/ThreadAccountantFactory.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.accounting;
+
+import org.apache.pinot.spi.env.PinotConfiguration;
+
+
+public interface ThreadAccountantFactory {
+  ThreadAccountant init(int numPqr, int numPqw, PinotConfiguration config);
```

Comment:
(nit) usually Factory provides `create` API

Reply:
numPqr -> numRunnerThreads
numPqw -> numWorkerThreads

Having this runner and worker concept in init also means that this can't be used on broker ?

Reply:
It can still be used on broker by simply setting numPqw to 0.

Reply:
Now `PerQueryCPUMemResourceUsageAccountant` only relies on total number of threads, as can be used for broker. Keeping interface this way for better extensibility.